rasterio/_io.pyx

```diff
@@ -828,17 +828,41 @@ cdef class DatasetReaderBase(DatasetBase):
         return sample_gen(self, xy, indexes)
 
 
+def is_zip(data):
+    """Detect zip compressed data"""
+    return data[:4].startswith(b'PK\x03\x04')
```

Comment:
@perrygeo I'm just going to delete that. I don't intend users to unknowingly create a `ZipMemoryFile`.